Client/Frontend/Browser/HomePanel/NTPDataSource.swift

```diff
@@ -0,0 +1,238 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+import Shared
+import BraveShared
+import BraveRewards
+
+class NTPDataSource {
```

Comment:
I renamed this class, so most of it didn't change, i will add comments on github where the code has changed